### DISCUSSION TITLE
Transfer NVMe volume

### DISCUSSION THREAD

**Discussion by Pompiedom** [007revad/Synology_enable_M2_volume#195] [Q&A] [answered]
Hi, my DS918+ crashed and I moved the NVMe drives to my new NAS (DS1821+). I run the M2 and HDD scripts but now I want to recover data from one of the M2 drives which I used as a fast docker storage.

I ran fdisk -l and I get

`Disk /dev/nvme0n1: 232.9 GiB, 250059350016 bytes, 488397168 sectors
Disk model: Samsung SSD 970 EVO Plus 250GB          
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: dos
Disk identifier: 0xb9024d1d

Device         Boot    Start       End   Sectors   Size Id Type
/dev/nvme0n1p1          8192  16785407  16777216     8G fd Linux raid autodetect
/dev/nvme0n1p2      16785408  20979711   4194304     2G fd Linux raid autodetect
/dev/nvme0n1p3      21241856 488395967 467154112 222.8G  f W95 Ext'd (LBA)
/dev/nvme0n1p5      21257952 488203231 466945280 222.7G fd Linux raid autodetect

Disk /dev/nvme1n1: 232.9 GiB, 250059350016 bytes, 488397168 sectors
Disk model: Samsung SSD 970 EVO Plus 250GB          
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: dos
Disk identifier: 0xf889c75e

Device         Boot    Start       End   Sectors   Size Id Type
/dev/nvme1n1p1          8192  16785407  16777216     8G fd Linux raid autodetect
/dev/nvme1n1p2      16785408  20979711   4194304     2G fd Linux raid autodetect
/dev/nvme1n1p3      21241856 488395967 467154112 222.8G  f W95 Ext'd (LBA)
/dev/nvme1n1p5      21257952 488203231 466945280 222.7G fd Linux raid autodetect`

How can I (if just for temporary) mount these so I can recover some data?

**Comment by Pompiedom** [comment, marked as answer]
I fixed this by following this great guide and was able to recover all data:
https://christopherkibble.com/posts/recovering-data-from-synology-mirror-on-new-nas/

1. Executed sudo fdisk -l to get a list of disks. I found /dev/sdc was my disk (size matched) and it had /dev/sdc1, /dev/sdc2, and /dev/sdc3 as devices on the list. I took note that /dev/sdc5 was the one I wanted as it was the largest of all of them so it's where my data would have lived.
2. Used mdadm to assemble the RAID (even though I only had half the mirror in there) by executing sudo mdadm --assemble --run /dev/md3 /dev/sdc5. I used /dev/md3 because md0, md1, and md2 were already in use.
3. Used nano to modify /etc/lvm/lvm.conf.  I commented out the line filter = [ "r|^/dev/md0$|", "r|^/dev/md1$|","a|/dev/md|", "r/.*/" ] and uncommented the line filter = [ "a|.*/|" ]. Without doing this, the disk wasn't showing up when I ran LVM commands.  If you installed nano, you can do this with sudo nano /etc/lvm/lvm.conf.
4. As both my old and new disks had identical volume group (VG) names which caused problems for me, I renamed the old one by first executing sudo vgdisplay, and then sudo lvm vgrename <uuid> old_disk where UUID is from the output of vgdisplay and 'old_disk' is actually what I called it so it would stand out. I then ran lvm vgchange -a y because an article I read mentioned you have to activate the VG after a name change. Honestly not sure if that was required, but no apparent harm.
5. Finally, I got the name of the logical volume (LV) by running sudo lvdisplay and making note that the LV Path (for me) was /dev/old_disk/lv. I was then able to mount it by creating a folder for it with sudo mkdir /mnt/old and then sudo mount /dev/old_disk/lv /mnt/old
6. With all of this complete, I can now browse my old file system in /mnt/old and copy data back to my volume by going into specific folders I wanted to copy data from and executing cp -R . /volume1/scratch where 'scratch' is the folder I created way up above in the pre-reqs.

**Reply by 007revad** [reply]
Nice.

That Chris Kibble blog has some interesting links in the credits.

**Comment by Patrick010** [comment]
Unfortunately the link is dead
